### PR TITLE
Support contribution instance build trigger

### PR DESCRIPTION
**Pull request by liorblob**
## XSOAR [Partner](https://xsoar.pan.dev/docs/partners/become-a-tech-partner)?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**.

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/27242

## Dependencies
https://github.com/demisto/contribution-management/pull/40

## Description
Support running a build for a contribution PR by number and head branch

## Does it break backward compatibility?
   - [X] Yes
       - Further details:
        TTL script will not be usable
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -316,14 +320,17 @@ references:
...
-                PACK=$(echo $CONTRIB_BRANCH | cut -d ":" -f 3)
-                echo 'Copy the changes from the contributor branch $USER/$BRANCH in the pack $PACK'
-                git remote add $USER git@github.com:$USER/content.git
-                git fetch $USER $BRANCH
-                rm -rf $PACK
-                git checkout $USER/$BRANCH $PACK
-                exit 0
-
+                python3 ./Utils/get_contrib_pack.py -p $PULL_REQUEST_NUMBER -b $BRANCH -u $USER
+  notify_contrib_pr: &notify_contrib_pr
```
Comment (Itay4) [inline .circleci/config.yml:324]:
```suggestion
  comment_instance_link_on_contrib_pr: &comment_instance_link_on_contrib_pr
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -892,6 +899,7 @@ jobs:
                 command: |
                   ./Tests/scripts/slack_notifier.sh 'test_playbooks' ./env_results.json
                 when: always
+            - *notify_contrib_pr
```
Comment (Itay4) [inline .circleci/config.yml:902]:
```suggestion
            - *comment_instance_link_on_contrib_pr
```

File: Utils/notify_pr.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
```
Comment (Itay4) [inline Utils/notify_pr.py:1]: rename script to `comment_on_pr.py`

**Comment by Itay4** [comment]
please add tests

File: Utils/notify_pr.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    comment_url = pr['comments_url']
+    headers = {'Authorization': 'Bearer ' + token}
+    response = requests.post(comment_url, json={'body': comment}, headers=headers, verify=False)
+    response.raise_for_status()
+
+    print_success('Successfully added the comment to the PR.')
+
+
+if __name__ == "__main__":
```
Comment (Itay4) [inline Utils/notify_pr.py:37]:
```suggestion
if __name__ == '__main__':
```

File: Utils/get_contrib_pack.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            print(f'Running command {command}')
+            run_command(command, is_silenced=False)
+    except Exception as e:
+        print_error(f'Failed to deploy contributed pack to base branch: {e}')
+        sys.exit(1)
+
+    print_success(f'Successfully updated the base branch with the contrib pack {pack_dir_name}')
+
+
+if __name__ == "__main__":
```
Comment (Itay4) [inline Utils/get_contrib_pack.py:62]:
```suggestion
if __name__ == '__main__':
```

File: Utils/notify_pr.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    response = requests.get(pr_url)
+    response.raise_for_status()
+    pr = response.json()
+    if not pr:
+        print_error('Could not find the pull request to reply on.')
+        sys.exit(1)
+
+    comment_url = pr['comments_url']
+    headers = {'Authorization': 'Bearer ' + token}
+    response = requests.post(comment_url, json={'body': comment}, headers=headers, verify=False)
```
Comment (Itay4) [inline Utils/notify_pr.py:31]: remove `verify=False`

File: Utils/get_contrib_pack.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+import requests
+from demisto_sdk.commands.common.tools import run_command, print_error, print_success
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Deploy a pack from a contribution PR to a branch')
+    parser.add_argument('-p', '--pr_number', help='Contrib PR number')
+    parser.add_argument('-b', '--branch', help='The contrib branch')
+    parser.add_argument('-u', '--user', help='The contrib repo')
```
Comment (Itay4) [inline Utils/get_contrib_pack.py:14]: update description

File: Utils/get_contrib_pack.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
```
Comment (Itay4) [inline Utils/get_contrib_pack.py:1]: rename script to `update_contribution_pack_in_base_branch`

File: .circleci/config.yml [hunk 9]
```diff
@@ -316,14 +320,17 @@ references:
...
+                python3 ./Utils/get_contrib_pack.py -p $PULL_REQUEST_NUMBER -b $BRANCH -u $USER
+  notify_contrib_pr: &notify_contrib_pr
+    when:
+      condition: << pipeline.parameters.contrib_branch >>
+      steps:
+        - run:
+            name: Comment on the contribution Pull Request
+            when: always
+            command: |
+              SERVER_URL=$(jq -r 'select(.[].Role == "Demisto Marketplace") | .[].InstanceDNS' ./env_results.json)
```
Comment (Itay4) [inline .circleci/config.yml:332]: add comment here that `Demisto Marketplace` is here to in order to refer to server master

**Review by yaakovi — APPROVED** [review]
looks ok to me

**Review by Itay4 — APPROVED** [review]

**Review by glicht — APPROVED** [review]

**Pull request merged**